tapir/coop/forms.py

```diff
@@ -247,3 +251,110 @@ class Meta:
         }
 
     share_owner = ShareOwnerChoiceField()
+
+
+class MembershipResignationForm(forms.ModelForm):
+    already_resigned = MembershipResignation.objects.values("share_owner")
+    in_three_years = _("Coop buys back share(s)")
+    cancellation_reason = forms.CharField(
+        max_length=1000,
+        widget=forms.Textarea(
+            attrs={"rows": 2, "placeholder": _("Please not more than 1000 characters.")}
+        ),
+    )
+    coop_buys_shares_back = forms.BooleanField(label=in_three_years, required=False)
+    share_owner = ShareOwnerChoiceField()
+    willing_to_gift_shares_to_coop = forms.BooleanField(
+        label="Willing to gift shares to coop", required=False
+    )
+    transfering_shares_to = ShareOwnerChoiceField(
+        required=False, label=_("Transfering share(s) to")
+    )
+
+    class Meta:
+        model = MembershipResignation
+        fields = [
+            "share_owner",
+            "cancellation_reason",
+            "cancellation_date",
+            "coop_buys_shares_back",
+            "willing_to_gift_shares_to_coop",
+            "transfering_shares_to",
+            "paid_out",
+        ]
+        widgets = {"cancellation_date": DateInputTapir()}
+
+    def clean(self):
```

Comment:
Could you split this in several functions?
Basically al if/elif blocks are independent from each other, right?

Then it allows you to give read-friendly names to each function, for example `if transfering_shares_to == share_owner:` can be in an `validate_gifter_and_giftee_not_the_same()` function or something similar.

Reply:
I separated now all validations in different defs. You asked probably 'cause of the sonar linter? Right?